File: go/palindrome_number.go
package algos

func isPalindrome(x int) bool {
	if x < 0 || (x%10 == 0 && x != 0) {
		return false
	} else if x < 10 {
		return true
	}
	reversed := 0
	for x > reversed {
		reversed = reversed*10 + x%10
		x /= 10
	}
	return reversed == x || x == reversed/10
}

File: rust/src/algos/palindrome_number.rs
pub fn is_palindrome(x: i32) -> bool {
    let mut number = x;
    if number < 0 || (number % 10 == 0 && x != 0) {
        false
    } else {
        let mut reversed = 0;
        while number > reversed {
            reversed = reversed * 10 + number % 10;
            number /= 10;
        }
        number == reversed || number == reversed / 10
    }
}
